### PR TITLE
[AIDAPP-207] Enhance the knowledge management experience by improving the article listing and article properties pages

### PR DESCRIPTION
**Pull request by ankit-canyon**
### Ticket(s) or GitHub Issue

- https://canyongbs.atlassian.net/browse/AIDAPP-207

### Technical Description

> Added created_at, updated_at column in Article properties view, Change the column orders in List Knowledge Base Items table. 

### Any deployment steps required?

> NA

### Are any Feature Flags Added?

> NA

_______________________________________________

### CODE REVIEW

File: app-modules/knowledge-base/src/Filament/Resources/KnowledgeBaseItemResource/Pages/ListKnowledgeBaseItems.php [hunk 1]
```diff
@@ -79,26 +81,37 @@ public function table(Table $table): Table
...
+                    ->toggleable()
                     ->sortable()
                     ->formatStateUsing(fn (bool $state): string => $state ? 'Yes' : 'No'),
-                TextColumn::make('category.name')
-                    ->label('Category')
-                    ->translateLabel()
-                    ->sortable(),
-                TextColumn::make('tags.name')
-                    ->badge()
-                    ->visible(fn (): bool => Feature::active('tags')),
```
Comment (Orrison) [inline app-modules/knowledge-base/src/Filament/Resources/KnowledgeBaseItemResource/Pages/ListKnowledgeBaseItems.php:101]: This column was removed here, but in the acceptance criteria it is still present, just modified and moved to be the last item in the list.

**Review by Orrison — CHANGES_REQUESTED** [review]

File: app-modules/knowledge-base/src/Filament/Resources/KnowledgeBaseItemResource/Pages/ListKnowledgeBaseItems.php (hunk 1 above)
Comment (ankit-canyon) [reply app-modules/knowledge-base/src/Filament/Resources/KnowledgeBaseItemResource/Pages/ListKnowledgeBaseItems.php:101]: Hi @Orrison ,

Tags (not default/not in filament column selector/not sortable)

But in Acceptance criteria it's mentioned as above. So by understanding that point isn't column should be removed from the list?

Comment (Orrison) [reply app-modules/knowledge-base/src/Filament/Resources/KnowledgeBaseItemResource/Pages/ListKnowledgeBaseItems.php:101]: Ah, that is a good point. Let me get clarity as to whether or not it should be removed.

File: app-modules/knowledge-base/src/Filament/Resources/KnowledgeBaseItemResource/Pages/ListKnowledgeBaseItems.php [hunk 2]
```diff
@@ -79,25 +81,40 @@ public function table(Table $table): Table
                     ->translateLabel()
                     ->searchable()
                     ->sortable(),
+                TextColumn::make('category.name')
+                    ->label('Category')
+                    ->translateLabel()
```
Comment (Orrison) [inline app-modules/knowledge-base/src/Filament/Resources/KnowledgeBaseItemResource/Pages/ListKnowledgeBaseItems.php:86]: Let's remove the `translateLabel` here and everywhere else in this page. We are getting rid of that as it is not needed.

File: app-modules/knowledge-base/src/Filament/Resources/KnowledgeBaseItemResource/Pages/ListKnowledgeBaseItems.php [hunk 3]
```diff
@@ -114,6 +131,34 @@ public function table(Table $table): Table
...
+                            ->format('d-m-Y'),
+                    ])
+                    ->query(function (Builder $query, array $data): Builder {
+                        return $query
+                            ->when(
+                                $data['created_after'],
+                                fn (Builder $query, $date): Builder => $query->whereDate('created_at', '>=', $date),
+                            );
+                    }),
+                Filter::make('updated_at')
```
Comment (Orrison) [inline app-modules/knowledge-base/src/Filament/Resources/KnowledgeBaseItemResource/Pages/ListKnowledgeBaseItems.php:148]: Let's add a test to `app-modules/knowledge-base/tests/KnowledgeBaseArticle/ListKnowledgeBaseArticlesTest.php` for each of these filters.

Creating KB articles, triggering the filter, then confirming that you do see that ones you would expect and not the ones that you would expect to not be listed.

**Review by Orrison — CHANGES_REQUESTED** [review]

**Pull request closed without merging**